saleor/core/taxes.py

```diff
@@ -82,3 +90,121 @@ class TaxData:
     shipping_price_net_amount: Decimal
     shipping_tax_rate: Decimal
     lines: List[TaxLineData]
+
+
+WEBHOOK_TAX_CODES_CACHE_KEY = "webhook_tax_codes"
+
+DEFAULT_TAX_CODE = "O9999999"
+DEFAULT_TAX_DESCRIPTION = "Unmapped Other SKU - taxable default"
+
+
+def _get_current_tax_app() -> Optional[App]:
+    """Return currently used tax app or None, if there aren't any."""
+    return (
+        App.objects.order_by("pk")
+        .for_event_type(WebhookEventSyncType.CHECKOUT_CALCULATE_TAXES)
+        .for_event_type(WebhookEventSyncType.ORDER_CALCULATE_TAXES)
+        .last()
+    )
+
+
+def _get_meta_code_key(app: App) -> str:
+    return f"{app.name}.code"
```

Comment:
It's not really building tax code per app, but rather per API provider. So when installing two different apps from the same provider, Saleor will use those keys. Also removing and adding the same app will not break integration, because API provider does not change.

I think using `App.identifier` here instead of `App.name` is a better choice.

Reply:
Right, sorry, I don't know why I've focused that `name` will behave in the same manner as `id` 😄